### ISSUE TITLE
OCaml 5 support

### ISSUE THREAD

**Issue by antoinemine** [nberth/mlgmpidl#17]
I see there have been some commits to help compiling under OCaml 5.

I've managed to compile master on OCaml 5.0.0 on the condition that profiling is disabled (it is enabled by default).
Should we disable profiling by default and do a new release and push to opam ?

Btw, I plan to make Apron compile under OCaml 5 and push a new opam version, which requires mlgmpidl, see https://github.com/antoinemine/apron/pull/66

**Comment by nberth** [comment]
Yes on the OCaml/C code side I think everything appears to be ready. The current issue is with integration in the opam repository: some changes are required by the repository maintainers in some `conf-*` packages, notably to use opam variables so as to offer cleaner support for new systems and architectures.

I've not had time to deal with that yet, so I'll try to get back to opam-repository maintainers and delay reworks on `conf-*` in the short term.

**Comment by nicokn** [comment]
> I've managed to compile master on OCaml 5.0.0 on the condition that profiling is disabled (it is enabled by default).
Should we disable profiling by default and do a new release and push to opam ?

Profiling is now, by default, disabled for all OCaml versions > 4.08.0 (PR https://github.com/nberth/mlgmpidl/pull/18)

**Comment by sim642** [comment]
> The current issue is with integration in the opam repository: some changes are required by the repository maintainers in some `conf-*` packages, notably to use opam variables so as to offer cleaner support for new systems and architectures.
> 
> I've not had time to deal with that yet, so I'll try to get back to opam-repository maintainers and delay reworks on `conf-*` in the short term.

Is it described anywhere what these changes are and why they are blocking mlgmpidl's OCaml 5 support? I couldn't find any reference (e.g. open issue) about it.
It would be great to have an OCaml 5 compatible version released at least for the systems and architectures that are already supported. Support for new architectures sounds like a separate matter from OCaml 5.

**Comment by nberth** [comment]
I finally managed to advance a bit on this front.
There's now a new version with tag 1.3.0 that should bring support for OCaml 5. Note the tag should still be considered temporary as I have probably some tweaks to tune that I'll discover using the opam CI. To do that last part I need https://github.com/ocaml/opam-repository/pull/23613 to be merged first.

**Comment by antoinemine** [comment]
I was wondering if there is any progress on having an opam release supporting OCaml 5.0 ?
I would like to make a new Apron release, with OCaml 5.0 support, and it seems that  the lack of mlgmpidl opam package prevents this CI to work...

**Comment by nberth** [comment]
> I was wondering if there is any progress on having an opam release supporting OCaml 5.0 ? I would like to make a new Apron release, with OCaml 5.0 support, and it seems that the lack of mlgmpidl opam package prevents this CI to work...

Alas the PR I made on the opam repository to tune the related `conf-*` packages have been stale for some time (cf https://github.com/ocaml/opam-repository/issues/23436). I think a quick temporary solution will be to first publish `mlgmpidl` on fewer architectures and systems until that issue is resolved.

**Comment by nberth** [comment]
@antoinemine Hopefully ocaml/opam-repository#24078 will get through soon enough.

**Comment by nberth** [comment]
@antoinemine ocaml/opam-repository#24078 is now merged, so you should be able to make a new release of Apron.

**Comment by nberth** [comment]
I'll wait for some returns on actual uses of the library with OCaml 5.0 (some uses of Apron for instance) before closing this issue.

**Comment by sim642** [comment]
I can confirm that Apron v0.9.14~beta.2 with mlgmpidl 1.2.15 works in Goblint.
